drivers/clock_control/clock_control_sam0_gclk.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2019 Derek Hageman <hageman@inthat.cloud>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <clock_control.h>
+
+struct sam0_gclk_config {
+	u32_t clock_frequency;
+	u8_t gen_id;
+};
+
+
+static int sam0_gclk_on(struct device *dev,
+			clock_control_subsys_t sub_system)
+{
+	const struct sam0_gclk_config *const cfg = dev->config->config_info;
+
+#ifdef GCLK_PCHCTRL_GEN
+	GCLK->PCHCTRL[(u32_t)sub_system].reg = GCLK_PCHCTRL_GEN(cfg->gen_id) |
+					       GCLK_PCHCTRL_CHEN;
+#else
+	GCLK->CLKCTRL.reg = GCLK_CLKCTRL_GEN(cfg->gen_id) |
+			    GCLK_CLKCTRL_CLKEN |
+			    GCLK_CLKCTRL_ID((u32_t)sub_system);
+#endif
+
```

Comment:
Do we need to wait here until the `SYNCBUSY` bit/register is cleared?

Reply:
As I read it: no.  The `CLKCTRL` or `PCHCTRL` registers are not listed as write-synchronized in the manuals so access shouldn't require anything special.  The various other parts of the GCLK control are, but it doesn't look like the actual gating requires synchronization.